src/config.py

```diff
@@ -96,75 +100,78 @@ class FlaskConfig(TypedDict):
-    # that are used to extend the validity of third party auth tokens
-    EXPLO_CLIENT: Mapping[str, str] = CLIENT.get("explo", {})
-    EXPLO_CLIENT_SECRET = EXPLO_CLIENT.get("secret", "")
-
-    OAUTH_CONF_URL = WEB_CLIENT.get("auth_uri", DEFAULT_OAUTH_CONF_URL)
-
-    # Analytics measurement id
-    MEASUREMENT_ID: str = j.get("MEASUREMENT_ID", "")
-    assert MEASUREMENT_ID, "MEASUREMENT_ID environment variable not set"
-
-    # Facebook app token, for login verification calls to the graph API
-    FACEBOOK_APP_ID: Mapping[str, str] = j.get("FACEBOOK_APP_ID", {})
-    FACEBOOK_APP_SECRET: Mapping[str, str] = j.get("FACEBOOK_APP_SECRET", {})
-    assert (
-        FACEBOOK_APP_SECRET
-    ), f"FACEBOOK_APP_SECRET not set correctly in {CLIENT_SECRET_FILE}"
-    assert FACEBOOK_APP_ID, f"FACEBOOK_APP_ID not set correctly in {CLIENT_SECRET_FILE}"
-
-    # Firebase configuration
-    FIREBASE_API_KEY: str = j.get("FIREBASE_API_KEY", "")
-    FIREBASE_SENDER_ID: str = j.get("FIREBASE_SENDER_ID", "")
-    FIREBASE_DB_URL: str = j.get("FIREBASE_DB_URL", "")
-    FIREBASE_APP_ID: str = j.get("FIREBASE_APP_ID", "")
-    assert (
-        FIREBASE_API_KEY
-    ), f"FIREBASE_API_KEY not set correctly in {CLIENT_SECRET_FILE}"
-    assert (
-        FIREBASE_SENDER_ID
-    ), f"FIREBASE_SENDER_ID not set correctly in {CLIENT_SECRET_FILE}"
-    assert FIREBASE_DB_URL, f"FIREBASE_DB_URL not set correctly in {CLIENT_SECRET_FILE}"
-    assert FIREBASE_APP_ID, f"FIREBASE_APP_ID not set correctly in {CLIENT_SECRET_FILE}"
-
-    # Apple ID configuration
-    APPLE_KEY_ID: str = j.get("APPLE_KEY_ID", "")
-    APPLE_TEAM_ID: str = j.get("APPLE_TEAM_ID", "")
-    APPLE_CLIENT_ID: str = j.get("APPLE_CLIENT_ID", "")
-    assert APPLE_KEY_ID, f"APPLE_KEY_ID not set correctly in {CLIENT_SECRET_FILE}"
-    assert APPLE_TEAM_ID, f"APPLE_TEAM_ID not set correctly in {CLIENT_SECRET_FILE}"
-    assert APPLE_CLIENT_ID, f"APPLE_CLIENT_ID not set correctly in {CLIENT_SECRET_FILE}"
-
-    # RevenueCat bearer token
-    RC_WEBHOOK_AUTH: str = j.get("RC_WEBHOOK_AUTH", "")
-
-
-# Read the Flask secret session key from file
-with open(os.path.join("resources", "secret_key.bin"), "rb") as f:
-    FLASK_SESSION_KEY = f.read()
-    assert len(FLASK_SESSION_KEY) == 64, "Flask session key is expected to be 64 bytes"
+try:
+    CLIENT_SECRET_FILE = {
+        "netskrafl": lambda: sm.get_json_secret("CLIENT_SECRET_NETSKRAFL"),
+        "explo-dev": lambda: sm.get_json_secret("CLIENT_SECRET_EXPLO"),
+        "explo-live": lambda: sm.get_json_secret("CLIENT_SECRET_EXPLO_LIVE"),
+    }.get(PROJECT_ID, lambda: sm.get_json_secret("DEFAULT_CLIENT_SECRET"))()
+except Exception:
+    CLIENT_SECRET_FILE = None
+
+assert CLIENT_SECRET_FILE, f"CLIENT_SECRET_FILE not set correctly for {PROJECT_ID}"
+
+# Read client secrets (some of which aren't really that secret) from the CLIENT_SECRET_FILE
+j = CLIENT_SECRET_FILE
+assert j is not None
```

Comment:
This assert is not needed, CLIENT_SECRET_FILE has already been tested for falsity